stackhpc_ipa_hardware_managers/system_nic.py

```diff
@@ -0,0 +1,241 @@
+# Copyright 2015 Rackspace, Inc.
+# Copyright 2018 StackHPC Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import glob
+import os
+
+from ironic_python_agent import errors
+from ironic_python_agent import hardware
+from ironic_python_agent import utils
+
+from oslo_concurrency import processutils
+from oslo_log import log
+
+UEVENT_PCI_SLOT_NAME_PREFIX = "PCI_SLOT_NAME="
+
+LOG = log.getLogger()
+
+
+def _get_lspci_output(vendor_id=None, device_id=None, **kwargs):
+    device_filter_arg = ''
+    if vendor_id or device_id:
+        device_filter_arg = "-d " + _get_device_filter_string(vendor_id,
+                                                              device_id)
+    lspci_output = ''
+    kwargs["shell"] = True
+    try:
+        lspci_output, _e = utils.execute(
+            "lspci -vmmnD {filter}".format(filter=device_filter_arg), **kwargs)
+    except (processutils.ProcessExecutionError, OSError):
+        raise errors.CleaningError("Unable to run lspci")
+    return lspci_output
+
+
+def _parse_lspci_output(lspci_output):
```

Comment:
nit: In text parsing code it's helpful to have a little description of the expected format, or a pointer to some docs.